include/zephyr/sys/util_internal_util_inc.h

```diff
@@ -0,0 +1,4113 @@
+/*
+ * Copyright (c) 2011-2014, Wind River Systems, Inc.
+ * Copyright (c) 2020, Nordic Semiconductor ASA
+ * Copyright (c) 2023, Meta
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_SYS_UTIL_INTERNAL_H_
+#error "This header should not be used directly, please include util_internal.h instead"
+#endif /* ZEPHYR_INCLUDE_SYS_UTIL_INTERNAL_H_ */
+
+#ifndef ZEPHYR_INCLUDE_SYS_UTIL_INTERNAL_UTIL_INC_H_
+#define ZEPHYR_INCLUDE_SYS_UTIL_INTERNAL_UTIL_INC_H_
+
+#define UTIL_INC_0 1
```

Comment:
this was already an issue prior to this patch but these `UTIL_INC_*` macros should actually be marked internal -- either use `Z_` prefix like for e.g `Z_LISTIFY`, or wrap in a ` @cond INTERNAL_HIDDEN` section.